ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/deduplication/HexOffsetSpec.scala

```diff
@@ -45,15 +45,19 @@ class HexOffsetSpec
           "00007a94" -> "00007a93",
         )
       ) { case (offset, expectedOffset) =>
-        HexOffset.firstBefore(
+        HexOffset.previous(
           Ref.HexString.assertFromString(offset)
         ) shouldBe Some(Ref.HexString.assertFromString(expectedOffset))
       }
     }
 
     "return lower offset" in forAll(offsets) { offset =>
-      val lowerOffset = HexOffset.firstBefore(offset).value
-      offset should be > [String] lowerOffset
+      if (offset.forall(_ == '0')) {
```

Comment:
Nit: I think it's clearer if you test the `BigInteger` value rather than the string form.